File: src/go/hello.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	msgTo := getInput("To")
	msgFrom := getInput("From")

	var msg = "This is " + msgFrom + "saying hello " + msgTo + "!"
	fmt.Println(msg)
}

func getInput(prompt string) string {
	reader := bufio.NewReader(os.Stdin)
	fmt.Print(prompt + ": ")
	input, _ := reader.ReadString('\n')
	return strings.TrimRight(input, "\r\n")
}

File: src/cpp/src/hello.cpp
#include <iostream>
#include <string>

using namespace std;

string getInput(string);

int main()
{
    string msgTo    = getInput("To");
    string msgFrom  = getInput("From");

    cout << "This is " << msgFrom << " saying hello " << msgTo << " from C++!" << endl;
}

string getInput(string prompt) 
{
    string input;
    cout << prompt << ": ";
    cin >> input;
    return input;
}
